src/aggregation/agg_req_with_accessor.rs

```diff
@@ -51,7 +53,8 @@ pub struct AggregationWithAccessor {
     /// If this needs to used by other aggregations, we need to refactor this.
     // NOTE: we can make all other aggregations use this instead of the `accessor` and `field_type`
     // (making them obsolete) But will it have a performance impact?
-    pub(crate) accessors: Vec<(Column<u64>, ColumnType)>,
+    pub(crate) accessors: HashMap<String, (Column<u64>, ColumnType)>,
```

Comment:
Because we need named access to fields. Alternatively we could leave the `accessors` as a vec and have a new `HashMap<String, ValueAccessor>` that could be used for the docvalue query. I think the second way can be useful because the query will otherwise need access to the segment reader to query term_id => Str/byte. More:
https://github.com/quickwit-oss/tantivy/pull/2198/commits/a26a3535a2275f88dcda679886385b79cdc415a9#diff-da09c2e16f884b4f3436a59897de837ff6b5b49003f54ed3e7d9e6687d7229d6R135-R146

Reply:
Why do we need the field name?

Reply:
- To map field name in docvalue_fields to a (accessor, type) pair
- Since we have 2 places — the aggregation itself and the field query, where we need the accessors, relying solely on order is not an option unless we have separate accessor vectors for each case.

Is there an alternative solution I might be missing?

Reply:
I created a new accessor field to hold `DynamicColumn` instead